Tests/MongoSwiftTests/DocumentTests.swift

```diff
@@ -305,126 +305,244 @@ final class DocumentTests: MongoSwiftTestCase {
         expect(doc["int64max"]).to(equal(.int64(Int64.max)))
     }
 
+    /// Test case that includes 'canonical' forms of BSON and Extended JSON that are deemed equivalent and may provide
```

Comment:
given how long this file already is, maybe consider pulling out the corpus tests into their own file?